src/odemis/acq/stitching/_registrar.py

```diff
@@ -645,10 +645,13 @@ def _get_shift(self, prev_tile, tile):
             return exp_tile_dist_px, 0
         # Normalized cross correlation (values between -1 and 1)
         ncc = (covar / (st_dev[0] * st_dev[1]))
-        overlap = numpy.abs(
-            numpy.subtract(tile.shape[::-1], numpy.abs(exp_tile_dist_px)))  # tile.shape is YX, so reverse it
-        if numpy.any(numpy.abs(meas_tile_dist_px) > overlap):
-            logging.info(f"Calculated shift {meas_tile_dist_px} is larger than the overlap size {overlap}, "
+        # tile.shape is YX, so reverse it
+        min_overlap = min(numpy.abs(numpy.subtract(tile.shape[::-1], numpy.abs(exp_tile_dist_px))))
+        # If the measured shift in x or y is higher than the minimum overlap, fall back to the expected shift.
+        # Use the minimum overlap because the overlapping parts are often elongated, even for elongated areas we do
+        # not expect a shift higher than the smallest side of the overlapping area.
+        if numpy.any(numpy.abs(meas_tile_dist_px) > min_overlap):
```

Comment:
Not really. Actually this is 2 comments.

First, what I mean is for now you're *guessing* that the "elongated areas" always mean that the two tiles are shifted from one another along the short dimension. That's correct in most cases, but I can image odd cases where the tile height and width are very different, and with a large overlap, this would cause the "elongated area" to be squarish and even possibly elongated in the other direction. It would not be a big deal, but it's also easy to *know* the dimension the tiles were shifted by just looking at exp_tile_dist_px (that's the largest axis, with the other axis being typically almost 0).

Secondly, you're now detecting a "bad shift estimation" by detecting that the shifts in both dimensions are > the overlap. Along the shifted dimension, that makes sense. However, for the other dimension (typically the elongated one), it's a very arbitrary value. Typically, it's expected to be near 0. So using the "short dimension" might still be too big. For instance, let's say the tiles are acquired with an overlap by 100px, with resolution of 1024x800px. So the overlapping area is 100x800px. X the maximum shift is 100px, that's fine. For the Y maximum shift, you could say it's 50% of 100px, or 150% of 100px, both would be arguable. It's quite arbitrary. For now you're using 100%. I think it's good enough, but it's worth clarifying in the comment that 100% is just an arbitrary number.